tests/e2e/specs/shopper/mini-cart.test.js

```diff
@@ -94,4 +123,89 @@ describe( 'Shopper → Mini Cart', () => {
 			);
 		} );
 	} );
+
+	describe( 'Filled mini cart', () => {
+		beforeAll( async () => {
+			await shopper.emptyCart();
+		} );
+
+		afterEach( async () => {
+			await shopper.emptyCart();
+		} );
+
+		it( 'The Mini Cart title shows correct amount', async () => {
+			await page.click(
+				'.wc-block-grid__product:first-child .add_to_cart_button'
+			);
+
+			await expect( page ).toMatchElement( '.wc-block-mini-cart__title', {
+				text: 'Your cart (1 item)',
+			} );
+
+			await page.mouse.click( 50, 200 );
+
+			await page.click(
+				'.wc-block-grid__product:last-child .add_to_cart_button'
+			);
+
+			await expect( page ).toMatchElement( '.wc-block-mini-cart__title', {
+				text: 'Your cart (2 items)',
+			} );
+		} );
+
+		it( 'The Mini Cart products table show added products', async () => {
+			const products = await page.$$(
+				'.wc-block-all-products .wc-block-grid__product'
+			);
+			if ( products.length === 0 ) {
+				return;
+			}
+
+			// Get a random product to ensure the test isn't false positive.
+			const product =
+				products[ Math.floor( Math.random() * products.length ) ];
```

Comment:
Does it have to be non deterministic here? Had random factor in tests bite me in the past :) How could this be a false postiive? Would it be because of cart not being properly cleared up in `beforeAll`?

Reply:
> Does it have to be non deterministic here?

It doesn't require but I want to replicate closer the real user behavior.

> How could this be a false positive?

`false positive` here may not exactly. But initially, I hard-coded the product title for the test. Lately, I get the product title dynamically, it can't be false positive anymore, I will update the comment to prevent confusion.